### PR TITLE
Checks if IP is in private space

### PR DESCRIPTION
**Pull request by asuweb**
Provides requested functionality in #449

---
This change is [Reviewable](https://reviewable.io/reviews/mailwatch/1.2.0/451)

### CODE REVIEW

**Comment by stefaweb** [comment]
For fr.php

'privatenetwork04' => '(Réseau privé)',

**Comment by stefaweb** [comment]
For fr.php:

```
'saspam03' => 'spam',
'sanotspam03' => 'N\'est pas un spam',
```

**Comment by asuweb** [comment]
Reviewed 6 of 8 files at r2, 2 of 2 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mailwatch/1.2.0/451)*

**Comment by asuweb** [comment]
Reviewed 2 of 8 files at r2.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mailwatch/1.2.0/451)*

**Comment by stefaweb** [comment]
Reviewed 6 of 8 files at r2, 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

*[mailscanner/languages/fr.php, line 183 at r3](https://reviewable.io:443/reviews/mailwatch/1.2.0/451#-Kba4sewKDzhzLS8MsTq:-Kba4sewKDzhzLS8MsTr:b-o8rwmm) ([raw file](https://github.com/mailwatch/1.2.0/blob/47577d78eb517a13ce835fd3e86e603a12734665/mailscanner/languages/fr.php#L183)):*
> 'saspam03' => 'spam',
>     'sanotspam03' => 'N\'est pas un spam',

 'saspam03' => 'Spam',
 'sanotspam03' => 'N\'est pas un Spam',

Sorry.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mailwatch/1.2.0/451)*

**Comment by Skywalker-11** [comment]
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[mailscanner/functions.php, line 3771 at r3](https://reviewable.io:443/reviews/mailwatch/1.2.0/451#-Kba4ZpZqOzLmV6_Sxfp:-Kba4ZpZqOzLmV6_Sxfq:br89v4e) ([raw file](https://github.com/mailwatch/1.2.0/blob/47577d78eb517a13ce835fd3e86e603a12734665/mailscanner/functions.php#L3771)):*
> ```PHP
>  * @return bool
>  */
> function privateNetwork($host)
> ```

suggest to rename the function to isPrivateNetwork to be more representative of its functionality (returning bool)

---

*[mailscanner/functions.php, line 3785 at r3](https://reviewable.io:443/reviews/mailwatch/1.2.0/451#-Kba5ccWHG18bzHpp-e0:-Kba5ccWHG18bzHpp-e1:bi7kdpo) ([raw file](https://github.com/mailwatch/1.2.0/blob/47577d78eb517a13ce835fd3e86e603a12734665/mailscanner/functions.php#L3785)):*
> ```PHP
>  * @return bool
>  */
> function cidr_match($ip, $cidr)
> ```

suggest to rename the function to isCidrMatch to be more representative of its functionality (returning bool)

---

*[mailscanner/languages/de.php, line 182 at r3](https://reviewable.io:443/reviews/mailwatch/1.2.0/451#-Kba6-vSnDEmTdEQeU3q:-Kba6-vSnDEmTdEQeU3r:be28t3w) ([raw file](https://github.com/mailwatch/1.2.0/blob/47577d78eb517a13ce835fd3e86e603a12734665/mailscanner/languages/de.php#L182)):*
> ```PHP
>     'auditlogupdateuser03' => 'Passwort für Benutzer aktualisiert',
>     'verifyperm03' => 'Bitte überprüfen Sie die Leseberechtigung für',
>     'saspam03' => 'spam',
> ```

'Spam'

---

*[mailscanner/languages/de.php, line 183 at r3](https://reviewable.io:443/reviews/mailwatch/1.2.0/451#-Kba5zw7q_u4eQExf1QO:-Kba5zw7q_u4eQExf1QP:b2sfsv3) ([raw file](https://github.com/mailwatch/1.2.0/blob/47577d78eb517a13ce835fd3e86e603a12734665/mailscanner/languages/de.php#L183)):*
> ```PHP
>     'verifyperm03' => 'Bitte überprüfen Sie die Leseberechtigung für',
>     'saspam03' => 'spam',
>     'sanotspam03' => 'not spam',
> ```

'Kein Spam'

---

*[mailscanner/languages/de.php, line 243 at r3](https://reviewable.io:443/reviews/mailwatch/1.2.0/451#-Kba5xnvSeLI6MQLloxy:-Kba5xnvSeLI6MQLloxz:bb6lkp8) ([raw file](https://github.com/mailwatch/1.2.0/blob/47577d78eb517a13ce835fd3e86e603a12734665/mailscanner/languages/de.php#L243)):*
> ```PHP
>     'geoipfailed04' => '(GeoIP Lookup fehlgeschlagen)',
>     'reversefailed04' => '(Reverse Lookup fehlgeschlagen)',
>     'privatenetwork04' => '(Private Network)',
> ```

'(Privates Netzwerk)'

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mailwatch/1.2.0/451)*

**Comment by asuweb** [comment]
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[mailscanner/functions.php, line 3771 at r3](https://reviewable.io:443/reviews/mailwatch/1.2.0/451#-Kba4ZpZqOzLmV6_Sxfp:-KbeluzsvoUIh8vniMST:bew2atr) ([raw file](https://github.com/mailwatch/1.2.0/blob/47577d78eb517a13ce835fd3e86e603a12734665/mailscanner/functions.php#L3771)):*
<details><summary><i>Previously, Skywalker-11 (Michael Pietsch) wrote…</i></summary><blockquote>

suggest to rename the function to isPrivateNetwork to be more representative of its functionality (returning bool)
</blockquote></details>

I agree - I'm going to change it to use the Pear packages for IPv4 & IPv6

---

*[mailscanner/functions.php, line 3785 at r3](https://reviewable.io:443/reviews/mailwatch/1.2.0/451#-Kba5ccWHG18bzHpp-e0:-Kbem0JxqYrXL6moxF42:bqog80t) ([raw file](https://github.com/mailwatch/1.2.0/blob/47577d78eb517a13ce835fd3e86e603a12734665/mailscanner/functions.php#L3785)):*
<details><summary><i>Previously, Skywalker-11 (Michael Pietsch) wrote…</i></summary><blockquote>

suggest to rename the function to isCidrMatch to be more representative of its functionality (returning bool)
</blockquote></details>

Agreed

---

*[mailscanner/languages/de.php, line 182 at r3](https://reviewable.io:443/reviews/mailwatch/1.2.0/451#-Kba6-vSnDEmTdEQeU3q:-Kbem2-REIKO5fASwQBp:b-zcjvte) ([raw file](https://github.com/mailwatch/1.2.0/blob/47577d78eb517a13ce835fd3e86e603a12734665/mailscanner/languages/de.php#L182)):*
<details><summary><i>Previously, Skywalker-11 (Michael Pietsch) wrote…</i></summary><blockquote>

'Spam'
</blockquote></details>

Thanks

---

*[mailscanner/languages/de.php, line 183 at r3](https://reviewable.io:443/reviews/mailwatch/1.2.0/451#-Kba5zw7q_u4eQExf1QO:-KbemBwLInfQUdgIowyO:b-zcjvte) ([raw file](https://github.com/mailwatch/1.2.0/blob/47577d78eb517a13ce835fd3e86e603a12734665/mailscanner/languages/de.php#L183)):*
<details><summary><i>Previously, Skywalker-11 (Michael Pietsch) wrote…</i></summary><blockquote>

'Kein Spam'
</blockquote></details>

Thanks

---

*[mailscanner/languages/de.php, line 243 at r3](https://reviewable.io:443/reviews/mailwatch/1.2.0/451#-Kba5xnvSeLI6MQLloxy:-KbemRS5yD7jT7aVErgP:bb6un1j) ([raw file](https://github.com/mailwatch/1.2.0/blob/47577d78eb517a13ce835fd3e86e603a12734665/mailscanner/languages/de.php#L243)):*
<details><summary><i>Previously, Skywalker-11 (Michael Pietsch) wrote…</i></summary><blockquote>

'(Privates Netzwerk)'
</blockquote></details>

Done

---

*[mailscanner/languages/fr.php, line 183 at r3](https://reviewable.io:443/reviews/mailwatch/1.2.0/451#-Kba4sewKDzhzLS8MsTq:-KbemQ-wjTzZxrttcdzl:bb6un1j) ([raw file](https://github.com/mailwatch/1.2.0/blob/47577d78eb517a13ce835fd3e86e603a12734665/mailscanner/languages/fr.php#L183)):*
<details><summary><i>Previously, stefaweb (Stéphane) wrote…</i></summary><blockquote>

> 'saspam03' => 'spam',
>     'sanotspam03' => 'N\'est pas un spam',

 'saspam03' => 'Spam',
 'sanotspam03' => 'N\'est pas un Spam',

Sorry.

</blockquote></details>

Done

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mailwatch/1.2.0/451)*

**Comment by asuweb** [comment]
Reviewed 9 of 9 files at r4.
Review status: all files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---

*[mailscanner/functions.php, line 3771 at r3](https://reviewable.io:443/reviews/mailwatch/1.2.0/451#-Kba4ZpZqOzLmV6_Sxfp:-KbqtK8BT_7ws64C0zkM:b39ozfm) ([raw file](https://github.com/mailwatch/1.2.0/blob/47577d78eb517a13ce835fd3e86e603a12734665/mailscanner/functions.php#L3771)):*
<details><summary><i>Previously, asuweb wrote…</i></summary><blockquote>

I agree - I'm going to change it to use the Pear packages for IPv4 & IPv6
</blockquote></details>

Re-written

---

*[mailscanner/functions.php, line 3785 at r3](https://reviewable.io:443/reviews/mailwatch/1.2.0/451#-Kba5ccWHG18bzHpp-e0:-KbqtLaa-DqM1c_GEyl_:b-896fix) ([raw file](https://github.com/mailwatch/1.2.0/blob/47577d78eb517a13ce835fd3e86e603a12734665/mailscanner/functions.php#L3785)):*
<details><summary><i>Previously, asuweb wrote…</i></summary><blockquote>

Agreed
</blockquote></details>

Done.

---

*[mailscanner/languages/de.php, line 182 at r3](https://reviewable.io:443/reviews/mailwatch/1.2.0/451#-Kba6-vSnDEmTdEQeU3q:-KbqtM_6GY8CJfsWzAQT:b-896fix) ([raw file](https://github.com/mailwatch/1.2.0/blob/47577d78eb517a13ce835fd3e86e603a12734665/mailscanner/languages/de.php#L182)):*
<details><summary><i>Previously, asuweb wrote…</i></summary><blockquote>

Thanks
</blockquote></details>

Done.

---

*[mailscanner/languages/de.php, line 183 at r3](https://reviewable.io:443/reviews/mailwatch/1.2.0/451#-Kba5zw7q_u4eQExf1QO:-KbqtNJkDAduQwkNAWxh:b-896fix) ([raw file](https://github.com/mailwatch/1.2.0/blob/47577d78eb517a13ce835fd3e86e603a12734665/mailscanner/languages/de.php#L183)):*
<details><summary><i>Previously, asuweb wrote…</i></summary><blockquote>

Thanks
</blockquote></details>

Done.

---

*[mailscanner/languages/de.php, line 243 at r3](https://reviewable.io:443/reviews/mailwatch/1.2.0/451#-Kba5xnvSeLI6MQLloxy:-KbqtO0L3Ao0TzOcovSJ:b-896fix) ([raw file](https://github.com/mailwatch/1.2.0/blob/47577d78eb517a13ce835fd3e86e603a12734665/mailscanner/languages/de.php#L243)):*
<details><summary><i>Previously, asuweb wrote…</i></summary><blockquote>

Done
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mailwatch/1.2.0/451)*

**Comment by asuweb** [comment]
Reviewed 2 of 2 files at r5.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mailwatch/1.2.0/451)*

**Comment by endelwar** [comment]
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[mailscanner/detail.php, line 53 at r5](https://reviewable.io:443/reviews/mailwatch/1.2.0/451#-KbsxshM1dGBYOpUQMCe:-KbsxshM1dGBYOpUQMCf:bda1wfs) ([raw file](https://github.com/mailwatch/1.2.0/blob/dd8a6c47ca2eca193f56eb46389e3cdd13ea2725/mailscanner/detail.php#L53)):*
> ```PHP
>     '192.168.0.0/16',
>     'fc00::/7',
>     'fd00::/8',
> ```

fc00::/7 contains fd00::/8

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mailwatch/1.2.0/451)*

File: mailscanner/lib/IPSet.php [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ * http://www.gnu.org/copyleft/gpl.html
+ *
+ * @file
+ * @author Brandon Black <blblack@gmail.com>
+ */
+#namespace IPSet;
```
Comment (endelwar) [inline mailscanner/lib/IPSet.php]: Why has namespace been commented?

**Comment by asuweb** [comment]
Review status: 7 of 9 files reviewed at latest revision, 2 unresolved discussions.

---

*[mailscanner/detail.php, line 53 at r5](https://reviewable.io:443/reviews/mailwatch/1.2.0/451#-KbsxshM1dGBYOpUQMCe:-KbvdVyVIdyrUbu5sg-l:b-oxgfcx) ([raw file](https://github.com/mailwatch/1.2.0/blob/dd8a6c47ca2eca193f56eb46389e3cdd13ea2725/mailscanner/detail.php#L53)):*
<details><summary><i>Previously, endelwar (Manuel Dalla Lana) wrote…</i></summary><blockquote>

fc00::/7 contains fd00::/8
</blockquote></details>

That's why I don't like IPv6 :-)

---

*[mailscanner/lib/IPSet.php, line 23 at r5](https://reviewable.io:443/reviews/mailwatch/1.2.0/451#-KbqtGC3fcMX9xP-s0rT-r5-23:-KbvddROZz0_IbMixSOD:b-886gsx) ([raw file](https://github.com/mailwatch/1.2.0/blob/dd8a6c47ca2eca193f56eb46389e3cdd13ea2725/mailscanner/lib/IPSet.php#L23)):*
<details><summary><i>Previously, endelwar (Manuel Dalla Lana) wrote…</i></summary><blockquote>

Why has namespace been commented?
</blockquote></details>

Fixed

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mailwatch/1.2.0/451)*

**Comment by asuweb** [comment]
Review status: 7 of 9 files reviewed at latest revision, 2 unresolved discussions.

---

*[mailscanner/detail.php, line 53 at r5](https://reviewable.io:443/reviews/mailwatch/1.2.0/451#-KbsxshM1dGBYOpUQMCe:-Kbves5MgbuFfs26-VhL:b-896fix) ([raw file](https://github.com/mailwatch/1.2.0/blob/dd8a6c47ca2eca193f56eb46389e3cdd13ea2725/mailscanner/detail.php#L53)):*
<details><summary><i>Previously, asuweb wrote…</i></summary><blockquote>

That's why I don't like IPv6 :-)
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mailwatch/1.2.0/451)*

**Comment by asuweb** [comment]
Reviewed 2 of 2 files at r6.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mailwatch/1.2.0/451)*

**Pull request merged**